topostats/run_topostats.py

```diff
@@ -228,16 +228,30 @@ def main(args=None):
 
     with Pool(processes=config["cores"]) as pool:
         results = defaultdict()
+        image_stats_all = defaultdict()
         with tqdm(
             total=len(img_files),
             desc=f"Processing images from {config['base_dir']}, results are under {config['output_dir']}",
         ) as pbar:
-            for img, result in pool.imap_unordered(
+            for img, result, individual_image_stats_df in pool.imap_unordered(
                 processing_function,
                 scan_data_dict.values(),
             ):
                 results[str(img)] = result
                 pbar.update()
+
+                # Add the dataframe to the results dict
+                image_stats_all[str(img)] = individual_image_stats_df
+
+                # Display completion message for the image
+                LOGGER.info(f"[{img.name}] Processing completed.")
+
+    LOGGER.info("Saving image stats dataframe to csv.")
```

Comment:
We could log where its saved to with...

```suggestion
    LOGGER.info(f"Saving image stats to : {config['output_dir']}/image_stats.csv.")
```